ee/agent/knapsack/knapsack.go

```diff
@@ -59,6 +62,18 @@ func New(stores map[storage.Store]types.KVStore, flags types.Flags, db *bbolt.DB
 	return k
 }
 
+// SetRunID sets the run ID in the knapsack
+func (k *knapsack) SetRunID(id string) {
```

Comment:
Do you need this method?

I'd probably do something like:
```
func (k *knapsack) GetRunID() string {
        if runID == "" {
            runID = ulid.New()
        }
	return runID
}
```

Reply:
I think that wouldn't cover the scenario where we want every time runLauncher is called to have a newID, I assume we could run into a situation where the ID is persisted if runLauncher is called after an update. Let me know if that's not the case and I can change this

Reply:
Chatted on slack -- I think this is worth testing. I'm not sure if the restart/exec would have access to the same memory

Reply:
I think if we did need to create a specific create method, my inclination would be to make `k.NewRunID()` and call it on startup. I think passing in the string feels wrong.

Reply:
I was able to test the scenario and after the update we do get a new ID when we generate the ID in the GET method. I refactored the code to not have a set.